### PR TITLE
Update Deployment reccomendations

### PR DESCRIPTION
**Pull request by Seanstoppable**
Rearrange so that the various 'levels' are grouped together.
Add internal tools that should probably be 'assessed'
Add legacy tools that should maybe be moved to Hold

### CODE REVIEW

File: recommendations/Deployment.md [hunk 1]
```diff
@@ -1,4 +1,28 @@
...
+  - [CodePipeline ](http://aws.amazon.com/codepipeline/)
+  - [EC2 Container Service](http://aws.amazon.com/ecs/)
+  - [Ionroller](https://github.com/gilt/ionroller/)
+
+    Ionroller is an exploration into 'immutable deploys'. It trades off speed of initial deploy (which involves new machines/elbs/etc being created) with the ability to: 
+      - test independently in production
+      - quickly roll over to a different stack, enabling [Blue Green Deploys](http://martinfowler.com/bliki/BlueGreenDeployment.html)
+
+  - [Mobile Service Tools](https://github.com/gilt/mobile-service-tools) (Closed Source)
+
```
Comment (gheine) [inline recommendations/Deployment.md]: We should add sbt-codedeploy https://github.com/gilt/sbt-codedeploy here too, as it's functionality somewhat overlaps with MST's while being open source.

**Comment by gheine** [comment]
Looks great!

**Comment by gheine** [comment]
+1

**Comment by andreyk0** [comment]
:+1:

**Pull request merged**